pkg/login/social/generic_oauth.go

```diff
@@ -130,7 +130,12 @@ func (s *SocialGenericOAuth) fillUserInfo(userInfo *BasicUserInfo, data *UserInf
 		userInfo.Email = s.extractEmail(data)
 	}
 	if userInfo.Role == "" {
-		userInfo.Role = s.extractRole(data)
+		role, err := s.extractRole(data)
+		if err != nil {
+			s.log.Error("Failed to extract role", "error", err)
```

Comment:
Please return the error here, unless there's a strong reason to swallow it.

Reply:
The reason to not return an error is that extracting a role is not a crucial thing for auth, if there's no role, default "Viewer" will be used.

Reply:
@alexanderzobnin is correct.

Reply:
@alexanderzobnin @marefr We should only log unexpected errors (i.e. not ones stemming from user mistakes) with error level, though.

Reply:
Yes, that makes sense. Maybe we should leave an old `searchJSONForAttr()` implementation with just changing log levels, like
```go
func (s *SocialBase) searchJSONForAttr(attributePath string, data []byte) string {
	if attributePath == "" {
		s.log.Debug("No attribute path specified")
		return ""
	}
	if len(data) == 0 {
		s.log.Debug("Empty user info JSON response provided")
		return ""
	}
	var buf interface{}
	if err := json.Unmarshal(data, &buf); err != nil {
		s.log.Error("Failed to unmarshal user info JSON response", "err", err.Error())
		return ""
	}
	val, err := jmespath.Search(attributePath, buf)
	if err != nil {
		s.log.Error("Failed to search user info JSON response with provided path", "attributePath", attributePath, "err", err.Error())
		return ""
	}
	strVal, ok := val.(string)
	if ok {
		return strVal
	}
	s.log.Debug("Attribute not found when searching JSON with provided path", "attributePath", attributePath)
	return ""
}
```

Reply:
I think we would achieve this best by only returning unexpected errors from `searchJSONForAttr`. The caller will then, for legacy reasons, only error log them.